Homework 12

letnasia opened the pull request:
Added .env support for database secrets
Added pydantic model to describe product update
Added queries to manipulate product (get/update/delete) Added API endpoints to manipulate product (get/put/delete)

Commit:
Homework 12

Added .env support for database secrets
Added pydentic model to describe product update
Added queries to manipulate product (get/update/delete)
Added API endpoints to manipulate product (get/put/delete)

p1v2 commented on app.py:
```diff
@@ -70,3 +72,39 @@ async def create_product(product_payload: ProductPayload, db: AsyncSession = Dep
+@app.put("/products/{product_id}", response_model=Product)
+async def update_product(
+        product_id: int,
+        product_update: ProductUpdate,
```
this model will not fit for PUT, because PUT requires all Product parameters to be passed

Commit:
Homework 12

Changed original API endpoint to PATCH
Added new proper PUT API endpoint that requires all fields

Commit:
Homework 12

Fixed duplicate function name
